pyproject.toml

```diff
@@ -7,6 +7,35 @@ requires = [
     "tomli; python_version < '3.11'",
 ]
 
+[project]
+name = "pyogrio"
+dynamic = ["version"]
+authors = [{ name = "Brendan C. Ward", email = "bcward@astutespruce.com" }]
+maintainers = [{ name = "pyogrio contributors" }]
+license = { text = "MIT License" }
+description = "Vectorized spatial vector file format I/O using GDAL/OGR"
+readme = "README.md"
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: MIT License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3",
+    "Topic :: Scientific/Engineering :: GIS",
+]
+requires-python = ">=3.8"
+dependencies = ["certifi", "numpy", "packaging"]
+
+[project.optional-dependencies]
+dev = ["cython"]
+test = ["pytest", "pytest-cov"]
+benchmark = ["pytest-benchmark"]
+geopandas = ["geopandas"]
+
+[project.urls]
+Home = "https://pyogrio.readthedocs.io/"
+Repository = "https://github.com/geopandas/pyogrio"
```

Comment:
Added both the doc page as the repo

Reply:
In the ["python packaging guide"](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#urls) there are some more urls listed... maybe list them all, or is that somehow overkill?

```
[project.urls]
Homepage = "https://example.com"
Documentation = "https://readthedocs.org"
Repository = "https://github.com/me/spam.git"
Issues = "https://github.com/me/spam/issues"
Changelog = "https://github.com/me/spam/blob/master/CHANGELOG.md"
```

Reply:
No strong opinion here, I just kept it consistent with geopandas repo for now